src/lib/Output/Generator/Json.php

```diff
@@ -4,209 +4,43 @@
  * @copyright Copyright (C) Ibexa AS. All rights reserved.
  * @license For full copyright and license information view LICENSE file distributed with this source code.
  */
+declare(strict_types=1);
 
 namespace Ibexa\Rest\Output\Generator;
 
 use Ibexa\Contracts\Rest\Output\Generator;
+use Ibexa\Rest\Output\Generator\Data\DataObjectInterface;
 
-/**
- * Json generator.
- */
 class Json extends Generator
 {
-    /**
-     * Data structure which is build during visiting;.
-     *
-     * @var array
-     */
-    protected $json;
-
-    /**
-     * Generator for field type hash values.
-     *
-     * @var \Ibexa\Rest\Output\Generator\Json\FieldTypeHashGenerator
-     */
-    protected $fieldTypeHashGenerator;
-
-    /**
-     * Keeps track if the document is still empty.
-     *
-     * @var bool
-     */
-    protected $isEmpty = true;
-
-    /**
-     * Enables developer to modify REST response media type prefix.
-     *
-     * @var string
-     */
-    protected $vendor;
-
-    /**
-     * @param \Ibexa\Rest\Output\Generator\Json\FieldTypeHashGenerator $fieldTypeHashGenerator
-     * @param string $vendor
-     */
-    public function __construct(Json\FieldTypeHashGenerator $fieldTypeHashGenerator, $vendor = 'vnd.ibexa.api')
-    {
+    public function __construct(
+        Json\FieldTypeHashGenerator $fieldTypeHashGenerator,
+        string $vendor = 'vnd.ibexa.api',
+    ) {
         $this->fieldTypeHashGenerator = $fieldTypeHashGenerator;
         $this->vendor = $vendor;
     }
 
-    /**
-     * Start document.
-     *
-     * @param mixed $data
-     */
-    public function startDocument($data)
-    {
-        $this->checkStartDocument($data);
-
-        $this->isEmpty = true;
-
-        $this->json = new Json\JsonObject();
-    }
-
-    /**
-     * Returns if the document is empty or already contains data.
-     *
-     * @return bool
-     */
-    public function isEmpty()
-    {
-        return $this->isEmpty;
-    }
-
-    /**
-     * End document.
-     *
-     * Returns the generated document as a string.
-     *
-     * @param mixed $data
-     *
-     * @return string
-     */
-    public function endDocument($data)
+    #[\Override]
+    public function endDocument(mixed $data): string
     {
-        $this->checkEndDocument($data);
+        parent::checkEndDocument($data);
```

Comment:
Why did you need to change this? Such syntax makes sense only if you are in the same method (e.g. would make sense if you were modifying `checkEndDocument` method instead of `endDocument`). Otherwise it should remain `$this`